### PR TITLE
initial for state related parts

### PR DESCRIPTION
**Pull request by jstavel**
I have created namespace rhsm.app.
It shares all states.

I have used mount library to solve dependecies states on states.
There is no need to move everything to rhsm.app at once.

Older code uses older rhsm.gui.tasks.ui

Code that depends on ui/windows state will use the new one.

BTW: I have prepared tests for this code (ie. tests of our testware :-)

### CODE REVIEW

File: src/rhsm/app/sm/ui.clj [hunk 1]
```diff
@@ -0,0 +1,12 @@
+(ns rhsm.app.sm.ui
+  (:require [mount.core :refer [defstate]]
+            [rhsm.gui.tasks.ui :as ui]
+            )
```
Comment (rarebreed) [inline src/rhsm/app/sm/ui.clj]: you can move these closing parens so they are all on line 3

File: src/rhsm/gui/tasks/ui.clj [hunk 2]
```diff
@@ -26,39 +27,43 @@ and returns a mapping like :registration-settings -> 'Registration Settings'"
...
-                :search-dialog "Searching"
-                ;;also does not exist anymore > tests have been moved to oldtests folder
-                ;:subscription-assistant-dialog "Subscription Assistant"
-                :subscription-redemption-dialog "Subscription Redemption"
-                :system-preferences-dialog "System Preferences"
-                :warning-dialog "Warning"
-                :repositories-dialog "manage_repositories_dialog"
-                :subscription-attachment-dialog "Subscription Attachment"}))
+(defmulti windows-map-by-family :family)
+(defmethod windows-map-by-family :default [release]
```
Comment (rarebreed) [inline src/rhsm/gui/tasks/ui.clj]: Add a newline here

File: src/rhsm/gui/tasks/ui.clj [hunk 3]
```diff
@@ -26,39 +27,43 @@ and returns a mapping like :registration-settings -> 'Registration Settings'"
...
+   :file-chooser "Select A File"
+   :filter-dialog "Filter Options"
+   :firefox-help-window "frmRedHatSubscriptionManagement-MozillaFirefox"
+   :firstboot-proxy-dialog "Proxy Configuration"
+   :firstboot-window "frm0"
+   :help-dialog "Subscription Manager Manual"
+   :import-dialog "Import Certificates"
+   :information-dialog "Information"
+   ;; renamed in 818238
+                                        ;:progress-dialog "Progress Dialog"
```
Comment (rarebreed) [inline src/rhsm/gui/tasks/ui.clj]: I wonder how all the comments got shifted over?

File: src/rhsm/gui/tasks/ui.clj [hunk 4]
```diff
@@ -26,39 +27,43 @@ and returns a mapping like :registration-settings -> 'Registration Settings'"
...
+   ;;also does not exist anymore > tests have been moved to oldtests folder
+                                        ;:subscription-assistant-dialog "Subscription Assistant"
+   :subscription-redemption-dialog "Subscription Redemption"
+   :system-preferences-dialog "System Preferences"
+   :warning-dialog "Warning"
+   :repositories-dialog "manage_repositories_dialog"
+   :subscription-attachment-dialog "Subscription Attachment"}
+  )
 
+(def windows (define-windows (define-windows (windows-map-by-family {:family :default
```
Comment (rarebreed) [inline src/rhsm/gui/tasks/ui.clj]: why (define-windows (define-windows...))?  Wouldn't it just be one?

```
(def windows (define-windows (windows-map-by-family ...)))
```

**Comment by jstavel** [comment]
I think I have used emacs indentation. No problem to switch the indentation off.

**Comment by jstavel** [comment]
mistake. I will fix it.

**Comment by jstavel** [comment]
I have fixed it. I am curios that I see commits that does belong to this pull request.

**Comment by jstavel** [comment]
BTW: I am not sure now that detailed should be multimethod. :family, variant and sm-version?

**Comment by jstavel** [comment]
I have fixed everything. New pull will arise.

**Pull request closed without merging**